check_api/src/main/java/com/google/errorprone/dataflow/nullnesspropagation/NullnessAnnotations.java

```diff
@@ -84,12 +89,13 @@ public static boolean annotationsAreAmbiguous(
   public static ImmutableList<AnnotationMirror> annotationsRelevantToNullness(
       Collection<? extends AnnotationMirror> annotations) {
     return annotations.stream()
-        .filter(a -> ANNOTATION_RELEVANT_TO_NULLNESS.test(simpleName(a).toString()))
+        .filter(NullnessAnnotations::isAnnotationRelevant)
         .collect(toImmutableList());
   }
 
   public static ImmutableList<AnnotationTree> annotationsRelevantToNullness(
       List<? extends AnnotationTree> annotations) {
+    // Missing support for FALSE_NULLNESS_ANNOTATIONS
```

Comment:
I'm not sure what to do about this or how much it matters. There are test cases that hit this method but none that hit the `MemberSelectTree` inside `simpleName(AnnotationTree)`. I can get a qualified name from a `MemberSelectTree` but I don't see how that can be accomplished for an `IdentifierTree`.

Reply:
That must be what `Matchers::isType` is for. This looks like it could work:

```java
  private static final Matcher<AnnotationTree> NOT_FALSE_NULLNESS_ANNOT =
      Matchers.not(
          Matchers.anyOf(FALSE_NULLNESS_ANNOTATIONS.stream().map(Matchers::isType).toList()));

  public static ImmutableList<AnnotationTree> annotationsRelevantToNullness(
      List<? extends AnnotationTree> annotations, final VisitorState state) {
    return annotations.stream()
        .filter(a -> NOT_FALSE_NULLNESS_ANNOT.matches(a, state))
        .filter(a -> ANNOTATION_RELEVANT_TO_NULLNESS.test(simpleName(a)))
        .collect(toImmutableList());
  }
```

but requires passing `VisitorState` to (one) `annotationsRelevantToNullness`.